libraries/chain/hotstuff/hotstuff.cpp

```diff
@@ -134,10 +138,10 @@ bool pending_quorum_certificate::add_weak_vote(const std::vector<uint8_t>& propo
    return true;
 }
 
-bool pending_quorum_certificate::add_vote(bool strong, const std::vector<uint8_t>& proposal_digest, size_t index,
+bool pending_quorum_certificate::add_vote(bool strong, size_t index,
                                           const bls_public_key& pubkey, const bls_signature& sig) {
-   return strong ? add_strong_vote(proposal_digest, index, pubkey, sig)
-                 : add_weak_vote(proposal_digest, index, pubkey, sig);
+   return strong ? add_strong_vote(index, pubkey, sig)
+                 : add_weak_vote(index, pubkey, sig);
```

Comment:
as discussed in the meeting, you'll have to pass the `finalizer_digest` from the `block_state`, and we can remove the `_proposal_digest member`